lib/msal-common/src/cache/CacheManager.ts

```diff
@@ -146,9 +155,15 @@ export abstract class CacheManager implements ICacheManager {
             return [];
         } else {
             const allAccounts = accountValues.map<AccountInfo>((value) => {
-                let accountObj: AccountEntity = new AccountEntity();
-                accountObj = CacheManager.toObject(accountObj, value) as AccountEntity;
-                return accountObj.getAccountInfo();
+                const accountEntity = CacheManager.toObject<AccountEntity>(new AccountEntity(), value);
+                const accountInfo = accountEntity.getAccountInfo();
+                const idToken = this.readIdTokenFromCache(this.clientId, accountInfo);
+                if (idToken && !accountInfo.idTokenClaims) {
+                    accountInfo.idTokenClaims = new AuthToken(idToken.secret, this.cryptoImpl).claims;
```

Comment:
I suppose `readIdTokenFromCach` only needs to run if `accountInfo.idTokenClaims` isn't already defined.

Reply:
I think it's good to run it each time in case claims change

Reply:
No. Would like to always read the token from cache. In case there is any change, two sources of truth??

Reply:
May not matter actually, since we are creating these account objects on the fly.